README.md

```diff
@@ -203,6 +203,15 @@ fabric --help
 
 ### Using the `fabric` client
 
+If you want to use it with OpenAI API compatible inference servers, such as [FastChat](https://github.com/lm-sys/FastChat), [Helmholtz Blablador](http://helmholtz-blablador.fz-juelich.de), [LM Studio](https://lmstudio.ai) and others, simply export the following environment variables:
+
+- `export OPENAI_BASE_URL=https://YOUR-SERVER:8000/v1/`
+- `export DEFAULT_MODEL="YOUR_MODEL"`
+
+And if your server needs authentication tokens, like Blablador does, you export the token the same way you would with OpenAI:
+  
+- `export OPENAI_API_KEY="YOUR TOKEN"`
+
```

Comment:
I tried exporting these variables, before running fabric, setting them to point to my local LM Studio server, and they only seem to be working with the `--gui`:

```shell script
export OPENAI_BASE_URL=http://localhost:1234/v1
export DEFAULT_MODEL="QuantFactory/Meta-Llama-3-8B-Instruct-GGUF"
export OPENAI_API_KEY=""

pbpaste | fabric --pattern improve_writing --stream
```

Errors with:
```
Error: Connection error.
Connection error.
```

But when I set the same environment variables and run with the gui parameter, it seems to start working:
```shell script
fabric --gui
# Then use the gui to input text
```

🤔

Reply:
Running the new small phi-3 Microsoft model via LM Studio

```text
env OPENAI_BASE_URL=http://localhost:1234/v1 fabric --listmodels
GPT Models:
Phi-3-mini-4k-instruct-fp16.gguf

Local Models:
llama2:latest
llama3:latest

Claude Models:
claude-3-opus-20240229
claude-3-sonnet-20240229
claude-3-haiku-20240307
claude-2.1
```

And now I did this:

```text
% env OPENAI_BASE_URL=http://localhost:1234/v1 fabric -m Phi-3-mini-4k-instruct-fp16.gguf --pattern raw_query
Answer these 2 questions:
1) Why is the sky blue?
2) What is the tallest tree in the world?
The color of the sky appears blue due to a phenomenon called Rayleigh scattering. As sunlight enters Earth's atmosphere, it collides with molecules and small particles present therein. These interactions cause shorter wavelength colors (blue and violet) to scatter more than longer wavelength colors (red and yellow). However, our eyes are more sensitive to blue light, which is why we perceive the sky as predominantly blue during daylight hours.

The tallest tree in the world is a coast redwood known as Hyperion. It stands at an impressive height of 379.7 feet (115.7 meters) and was discovered on August 25, 2006, by naturalists Chris Atkins and Michael Taylor.
```

But when I set OPENAI_API_KEY to blank like in your example, I get exactly the same error:

```text
% env OPENAI_API_KEY="" OPENAI_BASE_URL=http://localhost:1234/v1 fabric -m Phi-3-mini-4k-instruct-fp16.gguf --pattern raw_query
Make me a haiku about Large Language Models.
Error: Connection error.
Connection error.
```

Try it again without setting OPENAI_API_KEY to "" or set it to some random value.

```text
% env OPENAI_API_KEY="N/A" OPENAI_BASE_URL=http://localhost:1234/v1 fabric -m Phi-3-mini-4k-instruct-fp16.gguf --pattern raw_query
Make me a haiku about Large Language Models.
Silent minds awake,
Words flow like rivers in code,
AI's vast knowledge blooms.
```